src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,111 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private final float loadFactor = 0.75f;
+    private final int defaultCapacity = 16;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node<?, ?>[defaultCapacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int hash = hash(key);
+        int index = getIndex(hash);
+        Node<K, V> currentNode = table[index];
+        if (currentNode == null) {
+            table[index] = new Node<>(hash, key, value, null);
+            size++;
+        } else if (key == null) {
+            while (currentNode != null) {
+                if (currentNode.key == null) {
+                    currentNode.value = value;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+            currentNode = table[index];
+            table[index] = new Node<>(hash, key, value, currentNode);
+            size++;
+        } else {
+            while (currentNode != null) {
+                if (Objects.equals(currentNode.key, key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    currentNode.next = new Node<>(hash, key, value, null);
+                    size++;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
```

Comment:
Please combine. For us should be no difference if key is null or not.
if it's null then it's just going into first bucket but we anyway should check keys of existing nodes if their key is null or not so if we use null-safe comparison then logic is identical